components/tools/OmeroWeb/omeroweb/webgateway/decorators.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+#
+# Copyright (C) 2016 University of Dundee & Open Microscopy Environment.
+# All rights reserved.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""Decorators for use with the webgateway application."""
+
+import omero
+import omeroweb.decorators
+import logging
+import traceback
+from django.http import JsonResponse
+from functools import update_wrapper
+from api_exceptions import NotFoundError, BadRequestError, CreatedObject
+
+
+logger = logging.getLogger(__name__)
+
+
+class login_required(omeroweb.decorators.login_required):
+    """
+    webgateway specific extension of the OMERO.web login_required() decorator.
+    """
+
+    def on_not_logged_in(self, request, url, error=None):
+        """
+        Used for json api methods
+        """
+        return JsonResponse({'message': 'Not logged in'},
+                            status=403)
+
+
+class json_response(object):
+    """
+    Class-based decorator for wrapping Django views methods.
+    Returns JsonResponse based on dict returned by views methods.
+    Also handles exceptions from views methods, returning
+    JsonResponse with appropriate status values.
+    """
+
+    def __init__(self):
+        """Initialises the decorator."""
+        pass
+
+    # To make django's method_decorator work, this is required until
+    # python/django sort out how argumented decorator wrapping should work
+    # https://github.com/openmicroscopy/openmicroscopy/pull/1820
+    def __getattr__(self, name):
```

Comment:
Fix from @dpwrussell went into Django upstream in 1.7. Do we still need this?

Might be time, not in this PR, to roll back the hack added in #1820 as well.

/cc @aleksandra-tarkowska, @joshmoore